python/beeai_framework/errors.py

```diff
@@ -123,14 +123,17 @@ def explain(self) -> str:
         return "\n".join(output).strip()
 
     @classmethod
-    def ensure(cls, error: Exception) -> "FrameworkError":
+    def ensure(cls, error: Exception, *, message: str | None = None) -> "FrameworkError":
         if isinstance(error, FrameworkError):
             return error
 
         if isinstance(error, CancelledError):
             return AbortError(cause=error)
 
-        return cls(cause=error)
+        if message:
+            return cls(cause=error, message=message)
```

Comment:
IIUC this if None check is so we don't overwrite an existing message if there is not one passed in? Cause otherwise we shouldn't need to check and could just pass the None through